### ISSUE TITLE
Strange behavoir on discount items

### ISSUE THREAD

**Issue by jvkranenburg** [Onlineforce/whmcs-credit-invoice#1]
Thnx!! This is working great!

Only when a invoice has an taxed **discount** item this will fail. They are added as nagative invoiceitem.

The "add_hook('UpdateInvoiceTotal', 1, function($vars)" will calculate the tax wrongly. This should only be done on a "credit invoice".

Is it possible to change this?

**Comment by alexanderzinolidis** [comment]
@jvkranenburg Thank you! I will definitely investigate this behaviour more closely, we have not run in to invoices with taxed discounts yet. I'll get back to you with more info.

**Comment by zubfatal** [comment]
@jvkranenburg I know this is a very old issue, and development of this addon has probably been abandoned. I ran in to the same issue with discounts, where the overall tax for the invoice would be set to tax value of the actual discount.

A fix so far is to add this to the `UpdateInvoiceTotal` hook:
```php
add_hook('UpdateInvoiceTotal', 1, function($vars) {
	if (!invoice_is_credited($vars['invoiceid'])) {
		return false;
	}

	$invoice = Invoice::with('items')->findOrFail($vars['invoiceid']);
       ...
});
```
Another issue now, is that the created creditnote has the same `invoicenum` as the original invoice, which is bad, and thus far I haven't found a way to lookup/increment the next `invoicenum`.

**Comment by pschoenmakers** [comment]
@zubfatal I think the problem is that the amount of tax is calculated incorrectly

`$invoice->tax1 = $item->amount * ($invoice->taxRate1 / 100);`

should probably be changed to

`$invoice->tax1 = $invoice->subtotal * ($invoice->taxRate1 / 100);`

here: https://github.com/Onlineforce/whmcs-credit-invoice/blob/7dea95a7546c222eff346d4a9b8ab83461e66907/credit_invoice/hooks.php#L43

**Comment by alexanderzinolidis** [comment]
Thanks for looking into this guys.

Can you try the fix @pschoenmakers suggested and let me know if it works?

I will run some tests and update the addon as soon as I get some spare time.

**Comment by zubfatal** [comment]
I didn't try the fix @pschoenmakers  suggested, I've forked the code, and rewritten the credit function to use `localApi()` call, since WHMCS stores invoice data in multiple tables, not just `tblinvoices` and `tblinvoiceitems`

https://github.com/zubfatal/whmcs-credit-invoice

I've added configuration options for:
* Whether to negate the amount of the credit note (at least in Denmark, a credit note does not have negative amounts)
* Change the original invoice to `Cancelled`
* A "translation" of string to display on the invoice and credit note rather than displaying the actual note text
* Mark credit note as paid, using a given gateway (included)

Let me know if you want a pull request :)

**Comment by alexanderzinolidis** [comment]
@zubfatal Brilliant!

I will test your fork and you should definitely submit this as a PR. Thank you!

**Comment by alexanderzinolidis** [comment]
Closing since the changes were merged: #4
